### PR TITLE
Improved GaussLegendre and LegendreFactory

### PR DESCRIPTION
**Pull request by regislebrun**
Now, the computation of the nodes and weights of the quadrature rules are done using FastGL (https://sourceforge.net/projects/fastgausslegendrequadrature/), one of the fastest and most accurate algorithm to do so.

This PR implements a suggestion from @mbaudin47, thanks for it!

The code of FastGL is included into the OpenTURNS source-tree as:
  * the code has not changed on sourceforge since 2014
  * it is released under BSD license, see the web page
  * it is only one source code and one header

Here is an illustration of the improvement:
```python
import openturns as ot
import time

for iN in range(2**21):
    N = 2**iN
    print(f"GaussLegendre {N=}")
    t0 = time.time()
    algo = ot.GaussLegendre([N])
    nodes = algo.getNodes()
    weights = algo.getWeights()
    t1 = time.time()
    print("t=", t1 - t0, "s")
    if t1 > t0 + 1:
        break

factory = ot.LegendreFactory()
for iN in range(2**21):
    N = 2**iN
    print(f"LegendreFactory {N=}")
    t0 = time.time()
    nodes, weights = factory.getNodesAndWeights(N)
    t1 = time.time()
    print("t=", t1 - t0, "s")
    if t1 > t0 + 1:
        break
```
produces, with master:
```
GaussLegendre N=1
t= 4.601478576660156e-05 s
GaussLegendre N=2
t= 2.5510787963867188e-05 s
...
GaussLegendre N=1024
t= 0.7711801528930664 s
GaussLegendre N=2048
t= 6.972982168197632 s
LegendreFactory N=1
t= 1.1920928955078125e-05 s
LegendreFactory N=2
t= 6.9141387939453125e-06 s
...
LegendreFactory N=1024
t= 0.764716625213623 s
LegendreFactory N=2048
t= 6.909417152404785 s
```
and with this branch:
```
GaussLegendre N=1
t= 3.838539123535156e-05 s
GaussLegendre N=2
t= 9.5367431640625e-06 s
...
GaussLegendre N=8388608
t= 0.5979945659637451 s
GaussLegendre N=16777216
t= 1.189013957977295 s
LegendreFactory N=1
t= 0.0035042762756347656 s
LegendreFactory N=2
t= 1.1205673217773438e-05 s
...
LegendreFactory N=8388608
t= 0.8160159587860107 s
LegendreFactory N=16777216
t= 1.5231573581695557 s
```

### CODE REVIEW

File: ChangeLog [hunk 1]
```diff
@@ -1,4 +1,6 @@
 = New Features =
+ * New Gauss Legendre algorithm to generate weights and nodes, relying on
+	FastGL (https://sourceforge.net/projects/fastgausslegendrequadrature/)
```
Comment (jschueller) [inline ChangeLog:3]: this goes inside the 1.25 section, under Miscellaneous ?

File: COPYING.fastGL [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+//   quadrature nodes and weights. This makes it compatible with parallel computing paradigms.
+// - Accuracy: the error on the nodes and weights is within a few ulps (see the paper for details).
+
+// Disclaimer:
+// THIS SOFTWARE IS PROVIDED "AS IS" AND ANY EXPRESSED OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED 
+// WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE REGENTS OR 
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, 
+// BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+// HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR 
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
```
Comment (jschueller) [inline COPYING.fastGL:20]: also reference it our LICENSE file, into CMakeLists.txt, and finally in developer_guide/architecture.rst

File: LICENSE [hunk 3]
```diff
@@ -10,3 +10,4 @@ This library bundles several third-party codes with various licenses compatible
 - KissFFT (lib/src/Base/Algo/kissfft.hh), under BSD license, see COPYING.kissfft
 - KS distribution from Cephes library (lib/src/Uncertainty/Distribution/cephes/*) under BSD license, see COPYING.cephes
 - TNC optimization solver (lib/src/Base/Optim/algotnc.*) under Expat license, see COPYING.tnc
+- Gauss Legendre quadrature from FastGL library (lib/src/Base/Algo/fatgl*) under BSD license, see COPYING.fastGL
```
Comment (jschueller) [inline LICENSE:13]: fatgl

**Review by jschueller — APPROVED** [review]

File: LICENSE (hunk 3 above)
Comment (regislebrun) [reply LICENSE]: :-)

**Comment by jschueller** [comment]
@sofianehaddad could you review this one ?

**Review by sofianehaddad — APPROVED** [review]

**Pull request merged**